snuba/query/snql/parser.py

```diff
@@ -807,6 +807,42 @@ def parse(exp: Expression) -> Expression:
     query.transform_expressions(parse)
 
 
+def _mangle_query_aliases(
+    query: Union[CompositeQuery[QueryEntity], LogicalQuery],
+) -> None:
+    """
+    If a query has a subquery, the inner query will get its aliases mangled. This is
+    a problem because the outer query is using the inner aliases, not the inner
+    selected expression values.
+
+    So, we mangle the outer query column names to match the inner query aliases as well.
+    """
+
+    alias_prefix = "_snuba_"
+
+    def mangle_aliases(exp: Expression) -> Expression:
+        alias = exp.alias
+        if alias is not None:
+            return replace(exp, alias=f"{alias_prefix}{alias}")
```

Comment:
Can you have un-aliased expressions in the select clause of a subquery? You would need to generate an alias there otherwise the external query would not be able to reference them.
If we are relying on the parser behavior for that I would add this to the `validate_aliases` function so that we catch this before we go through query processing. Seeing such error later will be a lot harder to troubleshoot.

Specifically I would look into bare column expressions

Reply:
No we can't because we already alias every column with itself earlier. So every column should have an alias. As for other expressions, if you don't alias them then you can't reference them anyways so we don't need to worry about that.